src/elements/components/FeatheryAuthGate.tsx

```diff
@@ -0,0 +1,178 @@
+      // If user is not at the URL whitelisted for auth, take them there for login
+      history.replaceState(null, '', whitelistPath);
+    }
+
+    authState.clearLoader = () => setShowLoader(false);
+    // Register onLogin cb so it can be called by Client.submitAuthInfo
+    authState.onLogin = onLogin;
+    authState.onLogout = onLogout;
+
+    // If user passes authId as a prop, we need to submit it
+    if (authIdProp) {
+      const defaultClient = new Client();
+      defaultClient.submitAuthInfo({
+        authId: authIdProp
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!isAuthStytch()) return;
+
+    const authClient = getAuthClient();
+
+    const unsubscribe = authClient.session.onChange((newSession: any) => {
+      if (hasAuthedRef.current && newSession === null) {
+      } else if (newSession?.stytch_session?.session_jwt ?? getStytchJwt())
+        hasAuthedRef.current = true;
+    });
+
+    return unsubscribe;
+  }, [render]);
+
+  const onActive = useCallback(
+    throttle(
+      () => {
+        const stytchClient = getAuthClient();
+        if (!stytchClient) return;
+
+        const session = stytchClient.session.getSync();
+
+        if (session) {
+          stytchClient.session.authenticate({
+            session_duration_minutes: 1440
+          });
+        } else if (hasAuthedRef.current) {
+          stytchClient.session.revoke().then(logoutActions);
+        }
+      },
+      FIVE_MINUTES_IN_MILLISECONDS,
+      { leading: true, trailing: false }
+    ),
+    []
+  );
+
+  useIdleTimer({
+    // Need to use lodash throttle rather than IdleTimer throttle because the
+    // IdleTimer throttle resets after the idle timeout. So, after 10 seconds of
+    // no activity the user moves the mouse and onActive fires as expected. Then
+    // if the idle timeout passes again and the user moves the mouse again,
+    // onActive will fire a second time, only 10 seconds after the first time,
+    // despite the 5 minute throttle. We need a low timeout to quickly detect
+    // if the user session has expired upon returning to feathery.
+    onActive,
+    timeout: TEN_SECONDS_IN_MILLISECONDS
+  });
+
+  const form = (
+    <>
+      {showLoader && (
+        <div
+          style={{
+            // TODO: what should the background color be?
+            backgroundColor: '#FFF',
+            // backgroundColor: `#${activeStep?.default_background_color ?? 'FFF'}`,
+            position: 'fixed',
+            height: '100vh',
+            width: '100vw',
+            zIndex: 50,
+            display: 'flex',
+            justifyContent: 'center',
+            alignItems: 'center'
+          }}
+        >
+          <div style={{ height: '10vh', width: '10vh' }}>
+            <Spinner />
+          </div>
+        </div>
+      )}
+      <Form {...formProps} />
+    </>
+  );
+  if (!initInfo().authId || !formCompleted) {
+    return !fullPageLogin ? (
```

Comment:
IMO it might be safe for now to always assume fullPageLogin since i assume an auth gate will need to gate entire pages rather than sections of a page

Reply:
oh, yeah that's a fair point. was trying to not generalize too much from our case, but it does seem like a more safe default than I was thinking